app/models/driver.rb

```diff
@@ -0,0 +1,40 @@
+
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+
+  # must provide a name
+  validates :name, presence: {message: "%{value} must be given"}
+
+  # must provide a vin
+  validates :vin, presence: {message: "%{value} must be given"}
+
+  def self.select_driver
+    return Driver.all.sample
+  end
+
+  def average_rating
+    sum = 0
+    counter = 0
+    trips.each do |trip|
+      if trip.rating != nil
+        sum += trip.rating
+        counter += 1
+      end
+    end
+    if sum != 0
+      average = sprintf('%.2f', sum.to_f/counter)
+    else
+      average = nil
+    end
+    return average
+  end
+
+  def total_earnings
```

Comment:
This loop could also be written using [ActiveRecord calculation methods](http://api.rubyonrails.org/v5.1/classes/ActiveRecord/Calculations.html):
```ruby
def total_earnings
  trips.sum(:cost)
end
```